addon/mixins/disposable.js

```diff
@@ -57,46 +62,12 @@ export default Mixin.create({
   @public
   */
   registerDisposable(dispose) {
-    assert(
-      'Called `registerDisposable` where `dispose` is not a function',
-      typeof dispose === 'function'
-    );
-
-    let disposables = getOrAllocate(this, '_registeredDisposables', Array);
-    let disposable = toDisposable(dispose);
-
-    disposables.push(disposable);
-
-    return disposable;
+    return registerDisposable(this, dispose);
   },
 
   willDestroy() {
-    runDisposables(this._registeredDisposables);
+    runDisposables(registeredDisposables.get(this));
```

Comment:
Can we modify this to only need `runDisposables(this)`?  Basically, I think its a _super minor_ leak of the abstraction that `runDisposables` needs direct access to the weakmap here...

Reply:
Yes. We can totally do this.